File: src/main/resources/static/js/controller/MediaController.js
app.controller('MediaController', function($scope, $http) { 
	$http({
        method : "GET",
        url : "http://localhost:8080/medias"
    }).then(function mySuccess(response) {
        $scope.mediaList = response.data;
    }, function myError(response) {
        $scope.myError = response.statusText;
    });


    $scope.addMedia = function() {
    	$http({
        method : "POST",
        headers: {'Content-Type': 'application/json'},
        data :  $scope.media,
        url : "http://localhost:8080/medias"
    }).then(function mySuccess(response) {
        $scope.mediaList = response.data;

    }, function myError(response) {
        $scope.myError = response.statusText;
    });
    }
	
});

File: src/main/java/com/adrian/controller/MediaController.java
package com.adrian.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import com.adrian.dao.MediaDao;
import com.adrian.entity.Media;

@RestController
public class MediaController {
	
	@Autowired
	MediaDao mediaDao;
	
	@RequestMapping(value = "/medias", method = RequestMethod.GET, produces = { "application/JSON" })
	public List<Media> getAllMedia() {
		return mediaDao.getAllMedia();
	}
	
	@RequestMapping(value = "/medias", method = RequestMethod.POST, consumes = { "application/JSON" })
	public List<Media> addMedia(@RequestBody Media media) {
		int id = mediaDao.getAllMedia().size() + 1;
		media.setId(id);
		
		mediaDao.addMedia(media);
		
		return mediaDao.getAllMedia();
	}
}
